cmd/crdvalidator/README.md

```diff
@@ -0,0 +1,96 @@
+# crdvalidator
+
+## Summary
+A part of the core value proposition for RukPak is the safe upgrade of a [CustomResourceDefinition](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/) (`CRD`). As a result, this repository comes equipped with a "CRD Validator Webhook" (`crdvalidator`), that will validate all `CRD` upgrades created by RukPak. This protects your `BundleInstance` pivots from having potentially dangerous effects such as data loss. 
+
+### How to use
+
+`crdvalidator` will ensure that any `CRD` upgrade is safe as long as it is owned by RukPak. If you would like to disable validation for a `CRD` included in a specific `Bundle`, you will need to explicitly set the `core.rukpak.io/safe-upgrade-validation` annotation to `disabled` within that specific `Bundle`.
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    core.rukpak.io/safe-upgrade-validation: disabled # <--
+```
+
+### What is protected
+There are three main cases `crdvalidator` is checking against with every specified `CRD` upgrade. These are where a new `CRD`:
+1. Removes a stored version and ensures that removing the stored version does not result in data loss.
+2. Changes a version that old `CRD` had and it must validate existing CRs against the new schema.
+3. Adds a version that old `CRD` does not have. In this case `crdvalidator` checks if the conversion strategy is:
+    - None, then ensures that existing CRs validate with new schema.
+    - Webhook, then allow update (assuming webhook handles conversion correctly)
+
+#### Update the schema that invalidates existing CRs
+Say that you have installed a `CRD` onto the cluster with a Kind of `Sample` that has a single property of `examplearray`. 
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+    core.rukpak.io/safe-upgrade-validation: enabled
+  creationTimestamp: null
+  name: templates.combo.io
```

Comment:
nit: since in the description on L27 you reference "a CRD with a Kind of `Sample`" should the name of this CRD be something like `sample.example.io` instead?

Reply:
Good catch 👍🏻